scripts/download_data.py

```diff
@@ -318,50 +398,21 @@ def downloadFastas(self):
             referenceSetMetadata, referenceSetMetadataFilename)
         # Download chromosomes
         mkdirAndChdirList([self.referenceSetName])
-        cleanDir()
-        baseUrl = 'http://www.ebi.ac.uk/ena/data/view/'
         for chromosome in self.chromosomes:
-            accession = self.accessions[chromosome]
-            path = os.path.join(baseUrl, accession)
-            maxPos = self.chromMinMax.getMaxPos(chromosome)
-            minPos = 0
-            if self.excludeReferenceMin:
-                minPos = self.chromMinMax.getMinPos(chromosome)
-            args = urllib.urlencode({
-                'display': 'fasta',
-                'range': '{}-{}'.format(minPos, maxPos)})
-            url = '{}%26{}'.format(path, args)
-            tempFileName = '{}.fa.temp'.format(chromosome)
-            fileName = '{}.fa'.format(chromosome)
-            downloader = utils.HttpFileDownloader(url, tempFileName)
-            downloader.download()
-            # We need to replace the header on the downloaded FASTA
-            with open(tempFileName, "r") as inFasta,\
-                    open(fileName, "w") as outFasta:
-                # Write the new header
-                print(">{}".format(chromosome), file=outFasta)
-                inFasta.readline()
-                for line in inFasta:
-                    print(line, file=outFasta, end="")
-            os.unlink(tempFileName)
-            utils.log("Compressing {}".format(fileName))
-            utils.runCommand("bgzip {}".format(fileName))
-            compressedFileName = fileName + '.gz'
-            utils.log("Indexing {}".format(compressedFileName))
-            utils.runCommand("samtools faidx {}".format(compressedFileName))
-            # Assemble the metadata.
-            metadata = {
-                "md5checksum": getReferenceChecksum(compressedFileName),
-                "sourceUri": url,
-                "ncbiTaxonId": 9606,
-                "isDerived": False,
-                "sourceDivergence": None,
-                "sourceAccessions": [accession + ".subset"],
-            }
-            metadataFilename = "{}.json".format(chromosome)
-            dumpDictToFileAsJson(metadata, metadataFilename)
+            if isCheckpointFileExtant(chromosome):
+                logCheckpoint(chromosome)
+            else:
+                self._downloadFasta(chromosome)
+                createCheckpointFile(chromosome)
         escapeDir(3)
 
+    def cleanup(self):
+        utils.log('Removing checkpoint files')
+        for root, dirs, files in os.walk(os.getcwd()):
```

Comment:
Shouldn't this be the download directory instead of CWD? What if I ran this from my home directory, and I kept all my passwords in a .checkpoint file?